lib/distribute.c

```diff
@@ -494,20 +547,33 @@ int config_show_distribute(struct vty *vty)
 }
 
 /* Configuration write function. */
-int config_write_distribute(struct vty *vty)
+int config_write_distribute(struct vty *vty, const char *vrfname)
 {
 	unsigned int i;
 	int j;
 	int output, v6;
 	struct hash_backet *mp;
 	int write = 0;
+	const char *vrf_name;
 
-	for (i = 0; i < disthash->size; i++)
-		for (mp = disthash->index[i]; mp; mp = mp->next) {
+	if ((vrfname && strmatch(vrfname, VRF_DEFAULT_NAME)) ||
+	    !vrfname)
+		vrf_name = NULL;
+	else
+		vrf_name = vrfname;
+
+	for (i = 0; i < dist_ctx.disthash->size; i++)
+		for (mp = dist_ctx.disthash->index[i]; mp; mp = mp->next) {
 			struct distribute *dist;
 
 			dist = mp->data;
 
+			if ((dist->vrfname && !vrf_name) ||
+			    (!dist->vrfname  && vrf_name))
+				continue;
+			if (vrf_name && !strmatch(vrf_name, dist->vrfname))
+				continue;
```

Comment:
please drop all changes done in this function, and use `distribute_ctx->dist_ctx` instead of the global `dist_ctx`.